src/java.base/share/classes/java/lang/IO.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2024, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.lang;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOError;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * A collection of static methods that provide convenient access to {@link System#in}
+ * and {@link System#out} for line-oriented input and output.
+ * <p>
+ * The {@link #readln()} and {@link #readln(String)} methods in this class use internal
+ * objects that decode bytes read from {@code System.in} into characters. The charset used
+ * for decoding is specified by the {@link System#getProperties stdin.encoding} property.
+ * If this property is not present, or if the charset it names cannot be loaded, then
+ * UTF-8 is used instead. These internal objects are created upon the first call to
+ * either of the {@code readln} methods and are stored for subsequent reuse by these
+ * methods.
```

Comment:
@stuart-marks Can we rephrase this paragraph so that it doesn't use phrase "internal objects"?

Reply:
I'm in the midst of rewriting all of this stuff. The main concern here is that applications can use System.in more or less as they please (presumably without buffering) _until_ the first call to a `readln` method, at which time _something happens_ and effectively the application cannot use System.in anymore. I'll post the revision when I've finished it.

Reply:
Updated.